backend/src/main/java/com/redhat/cloud/notifications/routers/DrawerResource.java

```diff
@@ -0,0 +1,81 @@
+package com.redhat.cloud.notifications.routers;
+
+import com.redhat.cloud.notifications.db.Query;
+import com.redhat.cloud.notifications.db.repositories.DrawerNotificationRepository;
+import com.redhat.cloud.notifications.models.DrawerEntry;
+import com.redhat.cloud.notifications.routers.models.Meta;
+import com.redhat.cloud.notifications.routers.models.Page;
+import com.redhat.cloud.notifications.routers.models.PageLinksBuilder;
+import com.redhat.cloud.notifications.routers.models.UpdateNotificationDrawerStatus;
+import org.eclipse.microprofile.openapi.annotations.Operation;
+import org.jboss.resteasy.reactive.RestQuery;
+import javax.inject.Inject;
+import javax.validation.Valid;
+import javax.ws.rs.BeanParam;
+import javax.ws.rs.GET;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.SecurityContext;
+import javax.ws.rs.core.UriInfo;
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import static com.redhat.cloud.notifications.Constants.API_NOTIFICATIONS_V_1_0;
+import static com.redhat.cloud.notifications.routers.SecurityContextUtil.getOrgId;
+import static com.redhat.cloud.notifications.routers.SecurityContextUtil.getUsername;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+
+public class DrawerResource {
+
+    @Inject
+    DrawerNotificationRepository drawerRepository;
+
+    @Path(API_NOTIFICATIONS_V_1_0 + "/notifications/drawer")
+    public static class V1 extends DrawerResource {
+
+    }
+
+    @GET
+    @Produces(APPLICATION_JSON)
+    @Operation(summary = "Retrieve drawer notifications entries.", description =
+            "Allowed `sort_by` fields are `bundle`, `application`, `event`, `created` and read. The ordering can be optionally specified by appending `:asc` or `:desc` to the field, e.g. `bundle:desc`. Defaults to `desc` for the `created` field and to `asc` for all other fields."
+    )
+    public Page<DrawerEntry> getDrawerEntries(@Context SecurityContext securityContext, @Context UriInfo uriInfo,
+                                         @RestQuery Set<UUID> bundleIds, @RestQuery Set<UUID> appIds,
+                                         @RestQuery Set<UUID> eventTypeIds, @RestQuery LocalDateTime startDate, @RestQuery LocalDateTime endDate,
+                                         @RestQuery Boolean readStatus,
+                                         @BeanParam @Valid Query query) {
+
+        String orgId = getOrgId(securityContext);
+        String username = getUsername(securityContext);
+
+        List<DrawerEntry> drawerEntries = drawerRepository.getNotifications(orgId, username, bundleIds, appIds, eventTypeIds, startDate, endDate, readStatus,  query);
+
+        Long count = drawerRepository.count(orgId, username, bundleIds, appIds, eventTypeIds, startDate, endDate, readStatus);
```

Comment:
We could run the `count` query first and then skip the other one if `count == 0`.